plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/BaseSqlServerConnectorTest.java

```diff
@@ -167,11 +163,8 @@ public void testReadMetadataWithRelationsConcurrentModifications()
             // The test failure is not guaranteed
             // TODO (https://github.com/trinodb/trino/issues/10846): shouldn't fail
             assertThat(expected)
-                    .hasMessageMatching("(?s).*(" +
-                            "No task completed before timeout|" +
-                            "was deadlocked on lock resources with another process and has been chosen as the deadlock victim|" +
-                            // E.g. system.metadata.table_comments can return empty results, when underlying metadata list tables call fails
-                            "Expecting actual not to be empty).*");
+                    // before get columns from database metadata, for SqlServer we first try to get case sensitivity
+                    .hasMessageMatching(".*Failed to get case sensitivity for columns. Invalid object name.*");
```

Comment:
concurrent modification complaining about "Invalid object name" seems fishy. Can you share the full error message once?

Reply:
this is stack trace:
`2023-02-13T06:15:03.818-0600	ERROR	SplitRunner-9-204	io.trino.execution.executor.TaskExecutor	Error processing Split 20230213_121501_00021_uixjn.0.0.0-0 io.trino.connector.informationschema.InformationSchemaSplit@7cff7906 (start = 9.24334286868085E8, wall = 2011 ms, cpu = 0 ms, wait = 218 ms, calls = 1): JDBC_ERROR: Transaction (Process ID 54) was deadlocked on lock resources with another process and has been chosen as the deadlock victim. Rerun the transaction.
io.trino.spi.TrinoException: Transaction (Process ID 54) was deadlocked on lock resources with another process and has been chosen as the deadlock victim. Rerun the transaction.
	at io.trino.plugin.jdbc.BaseJdbcClient.getTableHandle(BaseJdbcClient.java:241)
	at io.trino.plugin.jdbc.ForwardingJdbcClient.getTableHandle(ForwardingJdbcClient.java:84)
	at io.trino.plugin.jdbc.jmx.StatisticsAwareJdbcClient.lambda$getTableHandle$3(StatisticsAwareJdbcClient.java:104)
	at io.trino.plugin.jdbc.jmx.JdbcApiStats.wrap(JdbcApiStats.java:35)
	at io.trino.plugin.jdbc.jmx.StatisticsAwareJdbcClient.getTableHandle(StatisticsAwareJdbcClient.java:104)
	at io.trino.plugin.jdbc.CachingJdbcClient.lambda$getTableHandle$4(CachingJdbcClient.java:290)
	at io.trino.collect.cache.EmptyCache.get(EmptyCache.java:97)
	at io.trino.plugin.jdbc.CachingJdbcClient.get(CachingJdbcClient.java:666)
	at io.trino.plugin.jdbc.CachingJdbcClient.getTableHandle(CachingJdbcClient.java:290)
	at io.trino.plugin.jdbc.CachingJdbcClient.lambda$getTableHandle$4(CachingJdbcClient.java:290)
	at com.google.common.cache.LocalCache$LocalManualCache$1.load(LocalCache.java:4868)
	at com.google.common.cache.LocalCache$LoadingValueReference.loadFuture(LocalCache.java:3533)
	at com.google.common.cache.LocalCache$Segment.loadSync(LocalCache.java:2282)
	at com.google.common.cache.LocalCache$Segment.lockedGetOrLoad(LocalCache.java:2159)
	at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2049)
	at com.google.common.cache.LocalCache.get(LocalCache.java:3966)
	at com.google.common.cache.LocalCache$LocalManualCache.get(LocalCache.java:4863)
	at io.trino.collect.cache.EvictableCache.get(EvictableCache.java:114)
	at io.trino.plugin.jdbc.CachingJdbcClient.get(CachingJdbcClient.java:666)
	at io.trino.plugin.jdbc.CachingJdbcClient.getTableHandle(CachingJdbcClient.java:290)
	at io.trino.plugin.jdbc.DefaultJdbcMetadata.listTableColumns(DefaultJdbcMetadata.java:689)
	at io.trino.spi.connector.ConnectorMetadata.streamTableColumns(ConnectorMetadata.java:302)
	at io.trino.plugin.base.classloader.ClassLoaderSafeConnectorMetadata.streamTableColumns(ClassLoaderSafeConnectorMetadata.java:311)
	at io.trino.plugin.base.classloader.ClassLoaderSafeConnectorMetadata.streamTableColumns(ClassLoaderSafeConnectorMetadata.java:94)
	at io.trino.metadata.MetadataManager.listTableColumns(MetadataManager.java:562)
	at io.trino.metadata.MetadataListing.listTableColumns(MetadataListing.java:164)
	at io.trino.connector.informationschema.InformationSchemaPageSource.addColumnsRecords(InformationSchemaPageSource.java:244)
	at io.trino.connector.informationschema.InformationSchemaPageSource.buildPages(InformationSchemaPageSource.java:212)
	at io.trino.connector.informationschema.InformationSchemaPageSource.getNextPage(InformationSchemaPageSource.java:179)
	at io.trino.operator.ScanFilterAndProjectOperator$ConnectorPageSourceToPages.process(ScanFilterAndProjectOperator.java:387)
	at io.trino.operator.WorkProcessorUtils$ProcessWorkProcessor.process(WorkProcessorUtils.java:413)
	at io.trino.operator.WorkProcessorUtils.getNextState(WorkProcessorUtils.java:262)
	at io.trino.operator.WorkProcessorUtils$YieldingProcess.process(WorkProcessorUtils.java:182)
	at io.trino.operator.WorkProcessorUtils$ProcessWorkProcessor.process(WorkProcessorUtils.java:413)
	at io.trino.operator.WorkProcessorUtils$3.process(WorkProcessorUtils.java:347)
	at io.trino.operator.WorkProcessorUtils$ProcessWorkProcessor.process(WorkProcessorUtils.java:413)
	at io.trino.operator.WorkProcessorUtils$3.process(WorkProcessorUtils.java:347)
	at io.trino.operator.WorkProcessorUtils$ProcessWorkProcessor.process(WorkProcessorUtils.java:413)
	at io.trino.operator.WorkProcessorUtils$3.process(WorkProcessorUtils.java:347)
	at io.trino.operator.WorkProcessorUtils$ProcessWorkProcessor.process(WorkProcessorUtils.java:413)
	at io.trino.operator.WorkProcessorUtils.getNextState(WorkProcessorUtils.java:262)
	at io.trino.operator.WorkProcessorUtils$BlockingProcess.process(WorkProcessorUtils.java:208)
	at io.trino.operator.WorkProcessorUtils$ProcessWorkProcessor.process(WorkProcessorUtils.java:413)
	at io.trino.operator.WorkProcessorUtils.lambda$flatten$6(WorkProcessorUtils.java:318)
	at io.trino.operator.WorkProcessorUtils$3.process(WorkProcessorUtils.java:360)
	at io.trino.operator.WorkProcessorUtils$ProcessWorkProcessor.process(WorkProcessorUtils.java:413)
	at io.trino.operator.WorkProcessorUtils$3.process(WorkProcessorUtils.java:347)
	at io.trino.operator.WorkProcessorUtils$ProcessWorkProcessor.process(WorkProcessorUtils.java:413)
	at io.trino.operator.WorkProcessorUtils.getNextState(WorkProcessorUtils.java:262)
	at io.trino.operator.WorkProcessorUtils.lambda$processStateMonitor$2(WorkProcessorUtils.java:241)
	at io.trino.operator.WorkProcessorUtils$ProcessWorkProcessor.process(WorkProcessorUtils.java:413)
	at io.trino.operator.WorkProcessorUtils.getNextState(WorkProcessorUtils.java:262)
	at io.trino.operator.WorkProcessorUtils.lambda$finishWhen$3(WorkProcessorUtils.java:256)
	at io.trino.operator.WorkProcessorUtils$ProcessWorkProcessor.process(WorkProcessorUtils.java:413)
	at io.trino.operator.WorkProcessorSourceOperatorAdapter.getOutput(WorkProcessorSourceOperatorAdapter.java:146)
	at io.trino.operator.Driver.processInternal(Driver.java:394)
	at io.trino.operator.Driver.lambda$process$8(Driver.java:297)
	at io.trino.operator.Driver.tryWithLock(Driver.java:689)
	at io.trino.operator.Driver.process(Driver.java:289)
	at io.trino.operator.Driver.processForDuration(Driver.java:260)
	at io.trino.execution.SqlTaskExecution$DriverSplitRunner.processFor(SqlTaskExecution.java:768)
	at io.trino.execution.executor.PrioritizedSplitRunner.process(PrioritizedSplitRunner.java:165)
	at io.trino.execution.executor.TaskExecutor$TaskRunner.run(TaskExecutor.java:523)
	at io.trino.$gen.Trino_testversion____20230213_121449_71.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: com.microsoft.sqlserver.jdbc.SQLServerException: Transaction (Process ID 54) was deadlocked on lock resources with another process and has been chosen as the deadlock victim. Rerun the transaction.
	at com.microsoft.sqlserver.jdbc.SQLServerException.makeFromDatabaseError(SQLServerException.java:265)
	at com.microsoft.sqlserver.jdbc.SQLServerResultSet$FetchBuffer.nextRow(SQLServerResultSet.java:5477)
	at com.microsoft.sqlserver.jdbc.SQLServerResultSet.fetchBufferNext(SQLServerResultSet.java:1790)
	at com.microsoft.sqlserver.jdbc.SQLServerResultSet.next(SQLServerResultSet.java:1048)
	at io.trino.plugin.jdbc.BaseJdbcClient.getTableHandle(BaseJdbcClient.java:228)
	... 66 more`

Reply:
but this error message doesn't contain "Invalid object name" so your change doesn't match this stack trace. 😕

Reply:
Ok as far as I understand this test:
we create and delete tables and concurrently perform read from system tables. Because when we perform read - we actually get table handle from database metadata - we can encounter this dead lock error.

But now, our flow is different, with case sensitivity, before read database metadata we actually perform SELECT *, and instead of concurrent dead lock error - we got Table not found, because it's atomic and table is already dropped or NOT.

Reply:
Ok I get it now. I was thinking too much about meaning of "Invalid" and didn't think it could also mean object not found. 😄 

Thanks for explaining.

Reply:
But previous one could also happen because table drop could occur after case sensitive read. So I think we need both cases